crates/host/src/oci.rs

```diff
@@ -214,7 +228,10 @@ impl Fetcher {
     /// Returns an error if either fetching fails or reading the fetched OCI path fails
     pub async fn fetch_component(&self, oci_ref: impl AsRef<str>) -> anyhow::Result<Vec<u8>> {
         let path = self
-            .fetch_path(oci_ref, vec![WASM_MEDIA_TYPE, OCI_MEDIA_TYPE])
+            .fetch_path(
+                oci_ref,
+                vec![WASM_MEDIA_TYPE, OCI_MEDIA_TYPE, WASM_LAYER_MEDIA_TYPE],
```

Comment:
Should these types be in `oci-wasm`?

Reply:
`WASM_MEDIA_TYPE` is a soon-to-be deprecated old type that exists in the oci-distribution crate (which it honestly shouldn't) that we have for backwards compatibility. I honestly don't know if we need `OCI_MEDIA_TYPE` any more since I don't think we pushed artifacts with that type, but I also kept it just in case